dashboard/api_edx_cache.py

```diff
@@ -0,0 +1,264 @@
+                updated_values = {
+                    'user': user,
+                    'course_run': course_run,
+                    'data': certificate.json,
+                }
+                models.CachedCertificate.objects.update_or_create(
+                    user=user,
+                    course_run=course_run,
+                    defaults=updated_values
+                )
+            # delete anything is not in the current certificates
+            models.CachedCertificate.delete_all_but(user, all_cert_course_ids)
+            # update the last refresh timestamp
+            cls.update_cache_last_access(user, cls.CERTIFICATE)
+        # submit a celery task to reindex the user
+        tasks.index_users.delay([user])
+
+    @classmethod
+    def update_cached_current_grades(cls, user, edx_client):
+        """
+        Updates cached current grade data.
+
+        Args:
+            user (django.contrib.auth.models.User): A user
+            edx_client (EdxApi): EdX client to retrieve enrollments
+        Returns:
+            None
+        """
+
+        course_ids = models.CachedEnrollment.active_course_ids(user)
+
+        # Current Grades are out of date, so fetch new data from edX.
+        current_grades = edx_client.current_grades.get_student_current_grades(
+            get_social_username(user), course_ids)
+
+        # the update must be done atomically
+        with transaction.atomic():
+            all_grade_course_ids = current_grades.all_course_ids
+            for course_run in CourseRun.objects.filter(edx_course_key__in=all_grade_course_ids):
+                current_grade = current_grades.get_current_grade(course_run.edx_course_key)
+                updated_values = {
+                    'user': user,
+                    'course_run': course_run,
+                    'data': current_grade.json,
+                }
+                models.CachedCurrentGrade.objects.update_or_create(
+                    user=user,
+                    course_run=course_run,
+                    defaults=updated_values
+                )
+            # delete anything is not in the current grades
+            models.CachedCurrentGrade.delete_all_but(user, all_grade_course_ids)
+            # update the last refresh timestamp
+            cls.update_cache_last_access(user, cls.CURRENT_GRADE)
+        # submit a celery task to reindex the user
+        tasks.index_users.delay([user])
+
+    @classmethod
+    def update_cache_if_expired(cls, user, edx_client, cache_type):
+        """
+        Checks if the specified cache type is expired and in case takes care to update it.
+
+        Args:
+            user (django.contrib.auth.models.User): A user
+            edx_client (EdxApi): EdX client to retrieve enrollments
+            cache_type (str): a string representing one of the cached data types
+        Returns:
+            None
+        """
+        cache_update_methods = {
```

Comment:
the reason because I put it here is because I cannot create a class variable referencing the class methods before defining the class methods.
The only way I know would be putting all the other methods in another class and then subclassing it to add this variable and this method.
what do you think?